x/leverage/keeper/collateral.go

```diff
@@ -7,23 +7,28 @@ import (
 	"github.com/umee-network/umee/v2/x/leverage/types"
 )
 
-// burnCollateral removes some uTokens from an account's collateral and burns them. This occurs
-// during liquidations.
-func (k Keeper) burnCollateral(ctx sdk.Context, addr sdk.AccAddress, coin sdk.Coin) error {
-	err := k.setCollateralAmount(ctx, addr, k.GetCollateralAmount(ctx, addr, coin.Denom).Sub(coin))
+// liquidateCollateral burns utoken collateral and sends the base token reward to the liquidator.
+// This occurs during direct liquidation.
+func (k Keeper) liquidateCollateral(ctx sdk.Context, borrower, liquidator sdk.AccAddress, utoken, token sdk.Coin) error {
+	err := k.setCollateralAmount(ctx, borrower, k.GetCollateralAmount(ctx, borrower, utoken.Denom).Sub(utoken))
 	if err != nil {
 		return err
 	}
-	if err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, sdk.NewCoins(coin)); err != nil {
+	if err = k.bankKeeper.BurnCoins(ctx, types.ModuleName, sdk.NewCoins(utoken)); err != nil {
 		return err
 	}
-	return k.setUTokenSupply(ctx, k.GetUTokenSupply(ctx, coin.Denom).Sub(coin))
+	if err = k.setUTokenSupply(ctx, k.GetUTokenSupply(ctx, utoken.Denom).Sub(utoken)); err != nil {
+		return err
+	}
+	return k.bankKeeper.SendCoinsFromModuleToAccount(
+		ctx, types.ModuleName, liquidator, sdk.NewCoins(token),
+	)
 }
 
-// removeCollateral removes some uTokens in fromAddr's collateral and sends them to toAddr. This
-// occurs when decollateralizing uTokens (in which case fromAddr and toAddr are the same) as well as
-// during liquidations, where toAddr is the liquidator.
-func (k Keeper) removeCollateral(ctx sdk.Context, fromAddr, toAddr sdk.AccAddress, coin sdk.Coin) error {
+// decollateralize removes fromAddrs' uTokens from circulation and redeems it for the underlying
+// token, sending it to the toAddr. It occurs when decollateralizing uTokens (in which case
+// fromAddr and toAddr are the same) as well as during liquidations, where toAddr is the liquidator.
```

Comment:
```suggestion
// decollateralize removes fromAddr's uTokens from the module, sending it to the toAddr.
// It occurs when decollateralizing uTokens (in which case fromAddr and toAddr are the
// same) as well as during liquidations, where toAddr is the liquidator.
```
The name decollateralize might be confusing since it happens during liquidations as well. removeCollateral also wasn't perfect, but I'm not sure of other names.

Reply:
the function is doing essentially decolaterizaiton, how other would you name action of going from utoken -> token.